### PR TITLE
switch our last two special cased admission plugins to config

### PR DESCRIPTION
**Pull request by deads2k**
We had two admission plugins that had special wiring to plumb configuration without using the "normal" admission config mechanism.  This removes those last two by creating the admission config types and then filling them in by synthetically injecting the values in.

@smarterclayton I think these were yours
@openshift/sig-master this is on the path to separate kubeapiserver launch in https://github.com/openshift/origin/pull/20578
/assign @mfojtik @sttts

### CODE REVIEW

File: pkg/service/admission/apis/externalipranger/install/install.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package install
```
Comment (mfojtik) [inline pkg/service/admission/apis/externalipranger/install/install.go]: @deads2k shouldn't we move all these apis into openshift/api ?

File: pkg/cmd/server/start/start_master.go [hunk 2]
```diff
@@ -448,6 +449,11 @@ func (m *Master) Start() error {
 			etcdserver.RunEtcd(m.config.EtcdConfig)
 		}
 
+		// convert the networkconfig to admissionconfig
```
Comment (sttts) [inline pkg/cmd/server/start/start_master.go]: nit: comment doesn't say more than the code line

File: pkg/service/admission/apis/externalipranger/doc.go [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// +k8s:deepcopy-gen=package,register
+
+// Package api is the internal version of the API.
```
Comment (sttts) [inline pkg/service/admission/apis/externalipranger/doc.go]: package externalipranger

File: pkg/service/admission/apis/externalipranger/v1/register.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package v1
+
+import (
+	"github.com/openshift/origin/pkg/service/admission/apis/restrictedendpoints"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// SchemeGroupVersion is group version used to register these objects
+var SchemeGroupVersion = schema.GroupVersion{Group: "", Version: "v1"}
```
Comment (sttts) [inline pkg/service/admission/apis/externalipranger/v1/register.go]: legacy group name for new apis?

File: pkg/service/admission/externalipranger/externalip_admission.go [hunk 5]
```diff
@@ -1,25 +1,66 @@
...
 	"net"
+	"reflect"
 	"strings"
 
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	admission "k8s.io/apiserver/pkg/admission"
 	kapi "k8s.io/kubernetes/pkg/apis/core"
+
+	"github.com/golang/glog"
```
Comment (sttts) [inline pkg/service/admission/externalipranger/externalip_admission.go]: order

File: pkg/service/admission/externalipranger/externalip_admission.go [hunk 6]
```diff
@@ -1,25 +1,66 @@
...
+				// should have been caught with validation
+				return nil, err
+			}
+
+			return NewExternalIPRanger(reject, admit, pluginConfig.AllowIngressIP), nil
 		})
 }
 
+func readConfig(reader io.Reader) (*externalipranger.ExternalIPRangerAdmissionConfig, error) {
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
```
Comment (sttts) [inline pkg/service/admission/externalipranger/externalip_admission.go]: oh, wow. we really still need the second case?

**Comment by sttts** [comment]
Looks good overall. Missing some test.

**Comment by deads2k** [comment]
> Looks good overall. Missing some test.

I think the previous unit and integration tests cover it.

File: pkg/service/admission/externalipranger/externalip_admission.go (hunk 6 above)
Comment (sttts) [reply pkg/service/admission/externalipranger/externalip_admission.go]: such a nil is a bug somewhere else.

Comment (deads2k) [reply pkg/service/admission/externalipranger/externalip_admission.go]: > such a nil is a bug somewhere else.

I won't argue it, but this is consistent with the rest of our admission plugins and I'm disinclined to change them at this juncture.

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**